### ISSUE TITLE
Issue: Act running on macos cannot resolve "v1" branch of actions/setup-java@v1

### ISSUE THREAD

**Issue by sophiederosa** [nektos/act#538]
## Act version

```none
act version 0.2.20
```

## Expected behaviour

I would expect Act to be able to locate the v1 tag, since it works fine in Actions

## Actual behaviour

Error: Unable to resolve v1: reference not found

## Workflow and/or repository

<details>
  <summary>workflow</summary>

```none
name: java-test
on:
  workflow_dispatch:
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: Set up JDK 1.11
      uses: actions/setup-java@v1
      with:
        java-version: '1.11'
```

</details>

## `act` output

<details>
  <summary>Log</summary>

```none
DEBU[0000] Loading environment from /Users/sophiederosa/Code/jenkout-fake-repo/.env 
DEBU[0000] Loading secrets from /Users/sophiederosa/Code/jenkout-fake-repo/.secrets 
DEBU[0000] Loading workflows from '/Users/sophiederosa/Code/jenkout-fake-repo/.github/workflows' 
DEBU[0000] Reading workflow '/Users/sophiederosa/Code/jenkout-fake-repo/.github/workflows/mega.yml' 
DEBU[0000] Planning event: workflow_dispatch            
DEBU[0000] Loading slug from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] Loading revision from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] HEAD points to 'abb8691c075f2e228a1e8befda75fab868846c43' 
DEBU[0000] HEAD matches refs/heads/mega-pipelines-testing 
DEBU[0000] using github ref: refs/heads/mega-pipelines-testing 
DEBU[0000] context env => map[ACT:true]                 
[java-test/build] 🚀  Start image=catthehacker/ubuntu:act-latest
DEBU[0000] Loading slug from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] Loading revision from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] HEAD points to 'abb8691c075f2e228a1e8befda75fab868846c43' 
DEBU[0000] HEAD matches refs/heads/mega-pipelines-testing 
DEBU[0000] using github ref: refs/heads/mega-pipelines-testing 
[java-test/build]   🐳  docker pull catthehacker/ubuntu:act-latest
DEBU[0000] Image exists? true                           
[java-test/build] Removed container: f7651b7b5f7af4ce9ce6e9276ad360cfef822444a4ad175551255df37a01d851
[java-test/build]   🐳  docker volume rm act-java-test-build
[java-test/build]   🐳  docker create image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[java-test/build] Created container name=act-java-test-build id=41e767be0e7574ae42ea5bd496bcdb771946ff197d3fb58165b48c31dbbd7e63 from image catthehacker/ubuntu:act-latest
[java-test/build] ENV ==> [RUNNER_TOOL_CACHE=/opt/hostedtoolcache RUNNER_OS=Linux RUNNER_TEMP=/tmp]
[java-test/build]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[java-test/build] Starting container: 41e767be0e7574ae42ea5bd496bcdb771946ff197d3fb58165b48c31dbbd7e63
[java-test/build] Started container: 41e767be0e7574ae42ea5bd496bcdb771946ff197d3fb58165b48c31dbbd7e63
DEBU[0000] Writing entry to tarball workflow/event.json len:2 
DEBU[0000] Writing entry to tarball workflow/envs.txt len:0 
DEBU[0000] Writing entry to tarball home/.act len:0     
[java-test/build] Extracting content to '/github/'
DEBU[0000] Loading slug from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] Loading revision from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] HEAD points to 'abb8691c075f2e228a1e8befda75fab868846c43' 
DEBU[0000] HEAD matches refs/heads/mega-pipelines-testing 
DEBU[0000] using github ref: refs/heads/mega-pipelines-testing 
DEBU[0000] Loading slug from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] Loading revision from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] HEAD points to 'abb8691c075f2e228a1e8befda75fab868846c43' 
DEBU[0000] HEAD matches refs/heads/mega-pipelines-testing 
DEBU[0000] using github ref: refs/heads/mega-pipelines-testing 
DEBU[0000] context env => map[ACT:true]                 
DEBU[0000] Loading slug from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] Loading revision from git directory '/Users/sophiederosa/Code/jenkout-fake-repo/.git' 
DEBU[0000] Found revision: abb8691c075f2e228a1e8befda75fab868846c43 
DEBU[0000] HEAD points to 'abb8691c075f2e228a1e8befda75fab868846c43' 
DEBU[0000] HEAD matches refs/heads/mega-pipelines-testing 
DEBU[0000] using github ref: refs/heads/mega-pipelines-testing 
DEBU[0000] context env => map[ACT:true]                 
DEBU[0000] context env => map[ACT:true CI:true GITHUB_ACTION:0 GITHUB_ACTIONS:true GITHUB_ACTOR:nektos/act GITHUB_API_URL:https://api.github.com GITHUB_ENV:/github/workflow/envs.txt GITHUB_EVENT_NAME:workflow_dispatch GITHUB_EVENT_PATH:/github/workflow/event.json GITHUB_GRAPHQL_URL:https://api.github.com/graphql GITHUB_REF:refs/heads/mega-pipelines-testing GITHUB_REPOSITORY:github/jenkout-fake-repo GITHUB_RUN_ID:1 GITHUB_RUN_NUMBER:1 GITHUB_SERVER_URL:https://github.com GITHUB_SHA:abb8691c075f2e228a1e8befda75fab868846c43 GITHUB_TOKEN: GITHUB_WORKFLOW:java-test GITHUB_WORKSPACE:/github/workspace HOME:/github/home INPUT_JAVA-VERSION:1.11] 
DEBU[0000] setupEnv: map[ACT:true CI:true GITHUB_ACTION:0 GITHUB_ACTIONS:true GITHUB_ACTOR:nektos/act GITHUB_API_URL:https://api.github.com GITHUB_ENV:/github/workflow/envs.txt GITHUB_EVENT_NAME:workflow_dispatch GITHUB_EVENT_PATH:/github/workflow/event.json GITHUB_GRAPHQL_URL:https://api.github.com/graphql GITHUB_REF:refs/heads/mega-pipelines-testing GITHUB_REPOSITORY:github/jenkout-fake-repo GITHUB_RUN_ID:1 GITHUB_RUN_NUMBER:1 GITHUB_SERVER_URL:https://github.com GITHUB_SHA:abb8691c075f2e228a1e8befda75fab868846c43 GITHUB_TOKEN: GITHUB_WORKFLOW:java-test GITHUB_WORKSPACE:/github/workspace HOME:/github/home INPUT_JAVA-VERSION:1.11] 
[java-test/build] ⭐  Run Set up JDK 1.11
[java-test/build]   ☁  git clone 'https://github.com/actions/setup-java' # ref=v1
[java-test/build]   cloning https://github.com/actions/setup-java to /Users/sophiederosa/.cache/act/actions-setup-java@v1
[java-test/build] Unable to resolve v1: reference not found
DEBU[0000] reference not found                          
DEBU[0000] reference not found                          
[java-test/build]   ❌  Failure - Set up JDK 1.11
DEBU[0000] reference not found                          
Error: reference not found
```

</details>

**Comment by lufia** [comment]
I will check this bug within next week.

**Comment by TheScriptingGuy** [comment]
I'm having the same issue! 
![image](https://user-images.githubusercontent.com/6164275/109174832-81fb0700-7785-11eb-9a45-aca2ce973f5b.png)

**Comment by catthehacker** [comment]
@TheScriptingGuy can you provide more info/repro?

**Comment by TheScriptingGuy** [comment]
@catthehacker it's basically the same error.

I've included the info below

#Act version
act version 0.2.20
#Expected behaviour
I would expect Act to be able to locate the v1 tag, since it works fine in Actions

#Actual behaviour
Error: Unable to resolve v1: reference not found

#Workflow and/or repository
workflow
#Steps to reproduce
##input (workflow)
```
name: CI
on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master

jobs:
  build:
    name: Build
    runs-on: ${{ matrix.os }}
    timeout-minutes: 10

    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest, macos-latest]

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-go@v2
        with:
          go-version: 1.15
      - run: make build
      - uses: actions/upload-artifact@v1
        with:
          name: podsync-${{ matrix.os }}
          path: bin/

  test:
    name: Test
    runs-on: ubuntu-latest
    timeout-minutes: 10

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-go@v1
        with:
          go-version: 1.15
      - env:
          VIMEO_TEST_API_KEY: ${{ secrets.VIMEO_ACCESS_TOKEN }}
          YOUTUBE_TEST_API_KEY: ${{ secrets.YOUTUBE_API_KEY }}
        run: make test

  checks:
    name: Checks
    runs-on: ubuntu-latest
    timeout-minutes: 10

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-go@v1
        with:
          go-version: 1.15

      - name: Go mod
        env:
          DIFF_PATH: "go.mod go.sum"
        run: |
          go mod tidy
          DIFF=$(git status --porcelain -- $DIFF_PATH)
          if [ "$DIFF" ]; then
            echo
            echo "These files were modified:"
            echo
            echo "$DIFF"
            echo
            exit 1
          fi

      - run: make lint

```

##act output

```
[Release/Publish] 🚀  Start image=catthehacker/ubuntu:act-latest
[CI/Test        ] 🚀  Start image=catthehacker/ubuntu:act-latest
[CI/Checks      ] 🚀  Start image=catthehacker/ubuntu:act-latest
[CI/Build-1     ] 🧪  Matrix: map[os:ubuntu-latest]
[CI/Build-3     ] 🚧  Skipping unsupported platform 'macos-latest'
[CI/Build-2     ] 🚧  Skipping unsupported platform 'windows-latest'
[CI/Build-1     ] 🚀  Start image=catthehacker/ubuntu:act-latest
[CI/Test        ]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[CI/Build-1     ]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[CI/Checks      ]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Release/Publish]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[CI/Test        ]   🐳  docker cp src=C:\Users\wessell\source\repos\podsync\podsync\. dst=\github\workspace
[CI/Build-1     ]   🐳  docker cp src=C:\Users\wessell\source\repos\podsync\podsync\. dst=\github\workspace
[Release/Publish]   🐳  docker cp src=C:\Users\wessell\source\repos\podsync\podsync\. dst=\github\workspace
[CI/Checks      ]   🐳  docker cp src=C:\Users\wessell\source\repos\podsync\podsync\. dst=\github\workspace
[CI/Test        ] ⭐  Run actions/checkout@v2
[Release/Publish] ⭐  Run git fetch --prune
[CI/Test        ]   ✅  Success - actions/checkout@v2
[CI/Checks      ] ⭐  Run actions/checkout@v2
[CI/Checks      ]   ✅  Success - actions/checkout@v2
[CI/Build-1     ] ⭐  Run actions/checkout@v2
[CI/Test        ] ⭐  Run actions/setup-go@v1
[CI/Test        ]   ☁  git clone 'https://github.com/actions/setup-go' # ref=v1
[CI/Build-1     ]   ✅  Success - actions/checkout@v2
[CI/Test        ] Unable to resolve v1: reference not found
[CI/Test        ]   ❌  Failure - actions/setup-go@v1
Error: reference not found
```

**Comment by sophiederosa** [comment]
Hey! Just wondering if there is any movement on this issue?

**Comment by sophiederosa** [comment]
Following up here! I think this issue would be fixed by this commit: nektos/act@14c06ee. However a new release hasn't been released since that code change went in. I built Act from source and it is working great for me locally, so @cplee if you could make a new release for Act I think it should be fixed!!

**Comment by ottumm** [comment]
I'm also running into this with act version 0.2.20 and the following workflow file:

```yaml
name: test
on:
  push:
    branches-ignore: [master]

jobs:
  lint:
    runs-on: ubuntu-20.04
    timeout-minutes: 5
    steps:
      - name: Checkout Code
        uses: actions/checkout@v2
        with:
          persist-credentials: false
      - name: Lint
        uses: github/super-linter@v3
        env:
          DEFAULT_BRANCH: main
```

**Comment by catthehacker** [comment]
Seems like this issue occurs when actions cache gets corrupted.
Running `rm -rf ~/.cache/act/` and retrying should fix the issue.

**Comment by catthehacker** [comment]
Fixed via #595
